packages/mobile/src/onboarding/registration/NameAndPicture.tsx

```diff
@@ -52,11 +56,22 @@ function NameAndPicture({}: Props) {
     ValoraAnalytics.track(OnboardingEvents.name_and_picture_set, {
       includesPhoto: false,
     })
-    dispatch(setName(nameInput))
+    dispatch(setName(newName))
+
+    // TODO: Store name and picture on CIP-8.
+    goToNextScreen()
   }
 
-  const onPhotoChosen = (photoData: string) => {
-    dispatch(setPicture(photoData))
+  const onPhotoChosen = (dataUrl: string | null) => {
+    if (!dataUrl) {
+      dispatch(setPicture(null))
+    } else {
+      const fileName = saveImageDataUrlToFile(
```

Comment:
What do you think about creating a wrapper function to save these photos, so we do not repeat this logic (especially path constructing)?